diff --git a/node_modules/yeoman-option-or-prompt/index.js b/node_modules/yeoman-option-or-prompt/index.js
index d7fd869..39e0ac6 100644
--- a/node_modules/yeoman-option-or-prompt/index.js
+++ b/node_modules/yeoman-option-or-prompt/index.js
@@ -12,39 +12,51 @@ module.exports = function(prompts) {
       // No option supplied, user will be prompted
       filteredPrompts.push(prompt);
     } else {
-      // Options supplied, add to props
+      // Options supplied, store it
       props[prompt.name] = normalize(option);
     }
   }, this);
 
-  if (filteredPrompts.length) {
-    return this.prompt(filteredPrompts).then(function mergeProps(mergeProps) {
-      // Merge mergeProps into props/
-      Object.assign(props, mergeProps);
-      return props;
-    });
+  if (!filteredPrompts.length) {
+    // No prompting required call the callback right away.
+    return Promise.resolve(props);
   }
 
-  // No prompting required call the callback right away.
-  return Promise.resolve(props);
+  return new Promise(async resolve => {
+    for (let i = 0; i < filteredPrompts.length; i++) {
+      let filteredPrompt = filteredPrompts[i];
+      var isWhenConditionFulfilled = typeof filteredPrompt.when !== 'function' ||
+        (typeof filteredPrompt.when === 'function' && filteredPrompt.when(props));
+      if (isWhenConditionFulfilled) {
+        delete filteredPrompt.when;
+        if (typeof filteredPrompt.choices === 'function') {
+          // We need to manually craft choices as YO's internal answers object cannot be modified
+          filteredPrompt.choices = filteredPrompt.choices(props);
+        };
+        const answers = await this.prompt(filteredPrompt);
+        Object.assign(props, answers);
+      }
+    }
+    resolve(props);
+  });
 };
 
 function normalize(option){
-    // TODO: 
-    // accept other types
+  // TODO:
+  // accept other types
 
-    if (typeof option === 'boolean') {
-      return option;
-    }
+  if (typeof option === 'boolean') {
+    return option;
+  }
 
-    if (typeof option === 'string'){
-      let lc = option.toLowerCase();
+  if (typeof option === 'string'){
+    let lc = option.toLowerCase();
 
-      // it's a boolean in string format
-      if (lc === 'true' || lc === 'false') {
-        return (lc === 'true');
-      } else {
-        return option;
-      }
+    // it's a boolean in string format
+    if (lc === 'true' || lc === 'false') {
+      return (lc === 'true');
+    } else {
+      return option;
     }
+  }
 }
